compiler/optimizer/src/main/java/org/apache/nemo/compiler/optimizer/pass/compiletime/annotating/DefaultScheduleGroupPass.java

```diff
@@ -113,7 +112,11 @@ public IRDAG apply(final IRDAG dag) {
       groupIdToVertices.putIfAbsent(curId, new ArrayList<>());
       groupIdToVertices.get(curId).add(irVertex);
 
-      final List<IREdge> allOutEdges = dag.getOutgoingEdgesOf(irVertex);
+      final List<IRVertex> verticesOfGroup = groupIdToVertices.get(curId);
+      final List<IREdge> allOutEdges = groupIdToVertices.get(curId).stream()
```

Comment:
allOutEdgesOfGroup?
Can you also leave a comment that this doesn't count edges between vertices in the group?